src/jit/importer.cpp

```diff
@@ -7309,6 +7309,13 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             /* only true object pointers can be virtual */
 
             assert(obj->gtType == TYP_REF);
+
+            // If we have a virtual call, is the class of 'this' a subtype of
+            // the class of the method? If so perhaps we can devirtualize.
+            if ((call->gtFlags & GTF_CALL_VIRT_KIND_MASK) == GTF_CALL_VIRT_VTABLE)
```

Comment:
You should also do this for `CORINFO_VIRTUALCALL_STUB`. It is used for interface calls under JIT and fragile NGen, and for all virtual calls under R2R. Thus, you should see some hits for R2R once you do that.